src/Groups/abelian_aut.jl

```diff
@@ -160,3 +159,117 @@ If `M` defines an endomorphism of `G`, return `true` if `M` defines an automorph
 """ 
 defines_automorphism(G::GrpAbFinGen, M::fmpz_mat) = isbijective(hom(G,G,M))
 
+
+
+
+################################################################################
+#
+#   Special functions for orthogonal groups of torsion quadratic modules
+#
+################################################################################
+
+
+"""
+    orthogonal_group(T::TorQuadMod, gensOT::Vector{fmpz_mat}) -> AutomorphismGroup{TorQuadMod}
+
+Return the subgroup of the orthogonal group of `G` generated by `gensOT`.
```

Comment:
This is (to me, coming from GAP) a really strange function. For me the more natural way to achieve the same would be: `sub(orthogonal_group(T), gensOT)`. But I assume there are reason why you can't / don't want to do it this way. What are they?

Reply:
Note that of course `orthogonal_group(T)` could be implemented lazily: i.e. it could do almost nothing beyond returning a "shell", and as long as nobody queries it for e.g. a concrete list of generators, no expensive computation would be performed.

Reply:
Okay, that was my concern because computing the generators can be a bit expensive.
Is there an example for such a shell? Then I can imitate that.

Reply:
As far as I can see that would imply a change to `Oscar.GAPGroup` namely to change `G.X` to `gap(G)` or something similar (which seems more reasonable to me)

Reply:
Oh and in some cases (e.g. the orthogonal group of an indefinite Z-lattice) I am not aware of any practical way to compute generators of the (infinite) "ambient" group much less how to do anything non-trivial with them. 
My way of thinking is similar to what we have with `matrix_group([some matrices])` and not `sub(GL(n,QQ),[some matrices])` since we do not necessarily care about the ambient group.

Reply:
I think we want the shallow `orthgonal_group(T)` eventually, but I believe this is outside the scope of this PR here. Maybe it could be `orthgonal_group([some endomorphisms of T])` then? But for me these are adjustments we can make later. Are there any objections @fingolfin?

Reply:
`orthogonal_group(T)` is allowed already. There is just this extra constructor. I can make it _private if needed.

Reply:
Mhhh o.k. maybe it is for the experts since I did not document which format the matrices need to be in to work.

Reply:
Maybe just mark it _private and we can merge this :D